modules/nw-ovn-kubernetes-live-migration.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+// * networking/openshift_sdn/rollback-to-ovn-kubernetes.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ovn-kubernetes-live-migration_{context}"]
+= Migrating to the OVN-Kubernetes network plugin by using the live migration method
+
+The following procedure entails checking for egress router resources and using the live migration method to migrate from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin.
+
+.Prerequisites
+
+* A cluster has been configured with the OpenShift SDN CNI network plugin in the network policy isolation mode.
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created a recent backup of the etcd database is available.
+* The cluster is in a known good state without any errors.
+* On all cloud platforms after updating software, a security group rule must be in place to allow UDP packets on port `6081` for all nodes.
+* Cluster administrators must remove any egress router resources.
+* You have reviewed the "Considerations for live migration to the OVN-Kubernetes network plugin" section of this document.
+
+.Procedure 
+
+. On the {product-title} web console, click *Observe* -> *Metrics*.
+
+. In the *Expression* box, add the following queries, then click *Add query*:
++
+[source,text]
+----
+network_attachment_definition_instances{networks="egress-router"} > 0
+----
+
+. Click *Run queries*. If any data points are found, you should not proceed with the OpenShift SDN to OVN-Kubernetes live migration procedure. These custom resources must be removed before using the live migration procedure.
+
+. To remove an egress router, you can delete the egress router custom resource definition:
++
+[source,terminal]
+----
+$ oc delete crd egressrouters.network.operator.openshift.io
+----
+
+ifdef::openshift-rosa,openshift-dedicated[]
+. Enter the following command to add the `unsupported-red-hat-internal-testing` annotation to the cluster-level network configuration:
```

Comment:
Just for clarification, this is only needed by managed clusters.

Reply:
Got it. The ifdef statements will ensure that this step is only in ROSA/OSD clusters. 

@pliurh regarding `$ oc delete crd egressrouters.network.operator.openshift.io`, this is what we did in the KCS. Is there another or different procedure required?

Reply:
The `egressrouters` is the OVN-K API for configuring egress router. In OpenShiftSDN, we use egress router pods instead. So we shall ask users to remove all the egress router pods before migration. https://docs.openshift.com/container-platform/4.14/networking/openshift_sdn/deploying-egress-router-layer3-redirection.html

Reply:
Can you check Step 1 and ensure its accuracy? cc @zhaozhanqi because it addresses https://github.com/openshift/openshift-docs/pull/76672#discussion_r1630962831 as well.

Reply:
ok for me
and maybe we can also provide how to find the egressrouter pod `oc get pods --all-namespaces -o json | jq '.items[] | select(.metadata.annotations."pod.network.openshift.io/assign-macvlan" == "true") | {name: .metadata.name, namespace: .metadata.namespace}'`

Reply:
@zhaozhanqi Updated to add that command here: https://76672--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#nw-ovn-kubernetes-live-migration_migrate-from-openshift-sdn

Should we tell users how to delete those pods here?